packages/admin/src/brevoConfiguration/BrevoConfigForm.tsx

```diff
@@ -106,47 +159,67 @@ export function BrevoConfigForm({ scope }: FormProps): React.ReactElement {
-
-        if (!sender) {
-            throw new Error("No sender selected");
-        }
-
-        const output = {
-            senderName: sender?.name,
-            senderMail: sender?.email,
-        };
-
         if (mode === "edit") {
             if (!data?.brevoConfig?.id) {
                 throw new Error("Missing id in edit mode");
             }
+
+            if (!data?.brevoConfig?.isApiKeySet) {
+                if (!state.apiKey) {
+                    throw new Error("Api key is required");
+                }
+
+                await client.mutate<GQLUpdateBrevoConfigMutation, GQLUpdateBrevoConfigMutationVariables>({
+                    mutation: updateBrevoConfigMutation,
+                    variables: { id: data?.brevoConfig?.id, input: { apiKey: state.apiKey }, lastUpdatedAt: data?.brevoConfig?.updatedAt },
+                });
+
+                return;
+            }
+
+            const sender = sendersData?.senders?.find((sender) => sender.id === state?.sender?.value);
+
+            if (!sender || !state.doiTemplate) {
+                throw new Error("Not all required fields are set");
+            }
+
+            const output = {
+                senderName: sender.name,
+                senderMail: sender.email,
+                doiTemplateId: Number(state.doiTemplate.value),
+            };
+
             await client.mutate<GQLUpdateBrevoConfigMutation, GQLUpdateBrevoConfigMutationVariables>({
                 mutation: updateBrevoConfigMutation,
                 variables: { id: data?.brevoConfig?.id, input: output, lastUpdatedAt: data?.brevoConfig?.updatedAt },
             });
         } else {
+            if (!state.apiKey) {
+                throw new Error("Api key is required");
+            }
             const { data: mutationResponse } = await client.mutate<GQLCreateBrevoConfigMutation, GQLCreateBrevoConfigMutationVariables>({
                 mutation: createBrevoConfigMutation,
-                variables: { scope, input: output },
+                variables: { scope, input: { apiKey: state.apiKey } },
             });
             if (!event.navigatingBack) {
                 const id = mutationResponse?.createBrevoConfig.id;
                 if (id) {
                     setTimeout(() => {
-                        stackSwitchApi.activatePage("edit", id);
+                        client.refetchQueries({ include: [namedOperations.Query.BrevoConfigForm] });
                     });
                 }
             }
         }
     };
 
-    if (error || senderError) throw error ?? senderError;
+    if (error || senderError || doiTemplatesError) throw error ?? senderError ?? doiTemplatesError;
 
-    if (loading || senderLoading) {
+    if (loading || senderLoading || doiTemplatesLoading) {
         return <Loading behavior="fillPageHeight" />;
     }
 
+    const isApiKeySet = mode !== "add" && data?.brevoConfig?.isApiKeySet;
```

Comment:
You could define this helper higher up and use it everywhere (instead of using data?.brevoConfig?.isApiKeySet directly)